Corona-Warn-App/src/main/java/de/rki/coronawarnapp/nearby/ENFClient.kt

```diff
@@ -0,0 +1,38 @@
+package de.rki.coronawarnapp.nearby
+
+import com.google.android.gms.nearby.exposurenotification.ExposureConfiguration
+import com.google.android.gms.nearby.exposurenotification.ExposureNotificationClient
+import de.rki.coronawarnapp.nearby.modules.diagnosiskeyprovider.DiagnosisKeyProvider
+import timber.log.Timber
+import java.io.File
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Suppress("DEPRECATION")
+@Singleton
+class ENFClient @Inject constructor(
+    private val googleENFClient: ExposureNotificationClient,
+    private val diagnosisKeyProvider: DiagnosisKeyProvider
+) : DiagnosisKeyProvider {
+
+    // TODO Remove this once we no longer need direct access to the ENF Client,
+    // i.e. in **[InternalExposureNotificationClient]**
+    internal val internalClient: ExposureNotificationClient
+        get() = googleENFClient
+
+    override suspend fun provideDiagnosisKeys(
+        keyFiles: Collection<File>,
+        configuration: ExposureConfiguration?,
+        token: String
+    ): Boolean {
+        Timber.tag(TAG).d(
+            "asyncProvideDiagnosisKeys(keyFiles=%s, configuration=%s, token=%s)",
+            keyFiles, configuration, token
+        )
+        return diagnosisKeyProvider.provideDiagnosisKeys(keyFiles, configuration, token)
+    }
+
+    companion object {
```

Comment:
Shouldn't companion objects be on top of the class !?

Reply:
While this is not always the case in current classes, new classes should put the companion object at the bottom. The goal is to follow the Kotlin styleguide: https://kotlinlang.org/docs/reference/coding-conventions.html